### PR TITLE
Added the 143 validation rule.

### LINKED ISSUE #32: Error code 143 - The reason for new episode code is not a valid code.
#### Error code
143
#### Description to display
The reason for new episode code is not a valid code.
#### DfE Long-form description of error
If completed reason for new episode must be coded S, P, L, T, U or B.
#### DfE suggested coding
If provided `<RNE>` must be a valid value  For allowed values please see associated code set.

For further info, refer to the DfE validation document [here](https://assets.publishing.service.gov.uk/government/uploads/system/uploads/attachment_data/file/974999/Children_looked_after_data_collection_2020_to_2021_validation_checks_-_version_1.3.pdf)

### PR DESCRIPTION
**Pull request by jfan2006**
Added the reason for new episode code checking for 143 validation rule. 

Closes #32

### CODE REVIEW

File: tests/test_validators.py [hunk 1]
```diff
@@ -1,6 +1,19 @@
...
+        'RNE': ['S', 'p', 'SP', 'a', '', pd.NA],
+    })
+
+    fake_dfs = {'Episodes': fake_data}
+
+    error_defn, error_func = validate_143()
+
+    result = error_func(fake_dfs)
+
+    assert result == {'Episodes': [1, 2, 3, 4, 5]}
```
Comment (mwaddoups) [inline tests/test_validators.py]: There's actually a subtlety to this rule that needs to be covered in the test and code (according to DfE). 

The description specifies "If provided <RNE> must be a valid value.".  Note the "if provided" - this means, that if it's not provided (i.e. it's pd.NA) , it shouldn't be a validation error. 

As such, I think this test case should just return `[1, 2, 3, 4]` as index 5 is valid.

File: validator903/validators.py [hunk 2]
```diff
@@ -1,6 +1,30 @@
...
+    )
+
+    def _validate(dfs):
+        if 'Episodes' not in dfs:
+            return {}
+        
+        episodes = dfs['Episodes']
+        code_list = ['S', 'P', 'L', 'T', 'B']
+
+        mask = episodes['RNE'].isin(code_list)
```
Comment (mwaddoups) [inline validator903/validators.py]: Similarly to my comment on the test, we need to also allow for cases where it is not provided. In this case you will want to combine this mask with another mask using the OR (`|`) operator. The relevant functions in pandas you'll need to check to see if it is missing are `.notna()` and `.isna()`.

See if you can get the test to pass by changing this.

**Review by mwaddoups — CHANGES_REQUESTED** [review]

File: tests/test_validators.py (hunk 1 above)
Comment (jfan2006) [reply tests/test_validators.py]: This is interesting! Totally understood your explanation about the DfE spec but I'm thinking in reality there shouldn't and won't any episode that doesn't have this reason code. (If it's before the end of June I'd try this on the DfE's SSDA903 system) 

I wondered I should contact DfE to double check/ clarify this validation rule... but for this project I'll make the change.

Comment (mwaddoups) [reply tests/test_validators.py]: Interesting - my thought is that you might see it missing for unclosed episodes at the end of the year? But that's off the top of my head, I can't remember for sure.

Comment (jfan2006) [reply tests/test_validators.py]: I think you were thinking about rule 148 - that's the reason episode ceased - the reason episode ceased could be blank (together with date episode ceased) if they were open/ unclosed episodes at the year end.

> Interesting - my thought is that you might see it missing for unclosed episodes at the end of the year? But that's off the top of my head, I can't remember for sure.

Comment (mwaddoups) [reply tests/test_validators.py]: Yep - you're right!

**Review by mwaddoups — APPROVED** [review]
All looks good now - nice work! Approving.

**Pull request merged**